visualiser/viewer.py

```diff
@@ -12,18 +15,152 @@
+        self.toggleShowNormalsAction.setChecked(False)
+
+        self.toggleShowWireAction = QAction('&Show Wires', self, checkable = True)
+        self.toggleShowWireAction.setToolTip('Toggle the wires')
+        self.toggleShowWireAction.triggered.connect(self.toggleShowWires)
+        self.toggleShowWireAction.setChecked(False)
+
+        self.toggleShowSampleAction = QAction('&Show Sample', self, checkable = True)
+        self.toggleShowSampleAction.setToolTip('Toggle the objects in prisms')
+        self.toggleShowSampleAction.triggered.connect(self.toggleShowSamples)
+        self.toggleShowSampleAction.setChecked(False)
+
+        self.toolbar = self.addToolBar('ToolBar')
+        self.toolbar.addAction(self.aboutAction)
+        self.toolbar.addAction(self.exitAction)
+        self.toolbar.addAction(self.toggleShowWireAction)
+        self.toolbar.addAction(self.toggleShowSampleAction)
+        self.toolbar.addAction(self.toggleShowNormalsAction)
+        self.toolbar.addAction(self.toggleShowAxesAction)
+
+        self.statusBar().showMessage('Ready') #only works for QMainWindow
+
+        # menubar = self.menuBar()
+        # fileMenu = menubar.addMenu('&Menu')
+        # fileMenu.addAction(exitAction)
+        #
+        # aboutMenu = menubar.addMenu('&About')
+        # aboutMenu.addAction(aboutAction)
+
+        #self.gl_widget = GLWidget()
+        #self.setCentralWidget(self.gl_widget)
+        #self.setCentralWidget(gl_widget)
+
+        self.setWindowTitle("GL Viewer")
+
+    def setGLWidget(self, gl_widget):
+        self.setCentralWidget(gl_widget)
+
+    def showAboutDlg(self):
+        about.show()
+
+    def changeProjection(self):
+        self._projectionType = (self._projectionType+1)%2
+        if self._projectionType:
+            self.changeProjectionAction.setText("Orthogonal")
+        else:
+            self.changeProjectionAction.setText("Perspective")
+
+        self.update()
+
+    def toggleShowAxes(self, checked = False):
+        self._showAxes = checked
+
+    def toggleShowNormals(self, checked = False):
+        self._showNormals = checked
+
+    def toggleShowWires(self, checked = False):
+        self._showWires = checked
 
-        self.gl_widget = GLWidget()
+    def toggleShowSamples(self, checked = False):
+        self._showSamples = checked
 
-        main_layout = QtWidgets.QHBoxLayout()
-        main_layout.addWidget(self.gl_widget)
+    @property
+    def projectionType(self):
+        return self._projectionType
 
-        self.setLayout(main_layout)
-        self.setWindowTitle("Hello GL")
+    @property
+    def showAxes(self):
+        return self._showAxes
+
+    @property
+    def showNormals(self):
+        return self._showNormals
+
+    @property
+    def showWires(self):
+        return self._showWires
+
+    @property
+    def showSamples(self):
+        return self._showSamples
+
+class AboutDlg(QWidget):
```

Comment:
Again, this seems like extra code for no reason. If you want to have instructions, add it to the `README.md`